### PR TITLE
Support groups-allowed-down-ratio

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

**Comment by hmusum** [comment]
@hakonhall for discussion

File: config-model/src/main/java/com/yahoo/vespa/model/content/ClusterControllerConfig.java [hunk 1]
```diff
@@ -122,79 +112,83 @@ public void getConfig(FleetcontrollerConfig.Builder builder) {
...
-                                           minNodeRatioPerGroup,
-                                           minSplitBits);
-    }
 
-}
+        private static Optional<Integer> maxGroupsAllowedDown(ModelElement tuning, boolean allowMoreThanOneContentGroupDown, int numberOfLeafGroups) {
+            var minGroupsUpRatio = tuning.childAsDouble("min-group-up-ratio");
+
+            if (minGroupsUpRatio != null) {
+                if (minGroupsUpRatio < 0.01 || minGroupsUpRatio > 1)
```
Comment (hakonhall) [inline config-model/src/main/java/com/yahoo/vespa/model/content/ClusterControllerConfig.java:159]: Why is  `0 < minGroupsUpRatio < 0.01`  disallowed?

File: config-model/src/main/resources/schema/content.rnc [hunk 2]
```diff
@@ -82,7 +82,7 @@ ClusterControllerTuning = element cluster-controller {
     element stable-state-period {  xsd:string { pattern = "([0-9\.]+)\s*([a-z]+)?" }  }? &
     element min-distributor-up-ratio { xsd:double }? &
     element min-storage-up-ratio { xsd:double }? &
-    element max-groups-allowed-down { xsd:nonNegativeInteger }?
+    element min-group-up-ratio { xsd:double }?
```
Comment (hakonhall) [inline config-model/src/main/resources/schema/content.rnc:85]: Hm, I checked with Henning:  Both he and I intuitively preferred the negated field, i.e. something like `groups-allowed-down-ratio`.  Henning preferred to leave out a `max-` prefix, and that seems slightly better to me too.

Comment (hakonhall) [reply config-model/src/main/resources/schema/content.rnc:85]: But now we see min-group-up-ratio matches the two other min-X-up-ratio names :thinking: .  The semantics of those may be sufficiently different that it confuse more than help?   I think the current min-X-up-ratio fields cause the cluster to go down, and is unrelated to orchestration?

Comment (hmusum) [reply config-model/src/main/resources/schema/content.rnc:85]: Naming was done in same style as the other `min-X-up-ratio`, but I agree that `groups-allowed-down-ratio` is easier to understand and work with, I will change the name

File: config-model/src/main/java/com/yahoo/vespa/model/content/ClusterControllerConfig.java (hunk 1 above)
Comment (hmusum) [reply config-model/src/main/java/com/yahoo/vespa/model/content/ClusterControllerConfig.java:159]: Will fix at the same time as changing the name

**Review by hakonhall — APPROVED** [review]

**Pull request merged**